tklinenums/tklinenums.py

```diff
@@ -116,90 +116,128 @@ def redraw(self, *_) -> None:
+            elif isinstance(widget_font, str):
+                # If user sent a tkinter Font instance
+                if "font" in widget_font:
+                    _font = Font(font=widget_font)
+
+                # If font family has 2+ words
+                elif "}" in widget_font:
+                    _family = widget_font.split("}")[0].replace("{", "")
+                    _size = widget_font.split("}")[1]
+                    _font = Font(family=_family, size=_size)
+
+                # If font family is just one word ("Consolas 20")
+                else:
+                    cur_font = widget_font.split(" ")
+                    _font = Font(family=cur_font[0], size=cur_font[1])
+
+            # If user is using customtkinter
+            # In customtkinter, fonts only accept tuple and CTkFont instance
+            elif isinstance(widget_font, tuple):
+                raise TypeError(
+                    "Tuple fonts type does not work correctly with the tilde parameter. Use a CTkFont() instance instead."
+                )
+
+            else:
+                _font = widget_font
+
             # Get the max amount of lines that can fit in the textwidget
             max_lines = self.textwidget.winfo_height() // _font.metrics()["linespace"]
 
-            # if the last line is greater than max_visible_lines, you won't need a tilde char
-            if int(last_line) < max_lines:
-            # If user send a tilde parameter, the loop range will change to the max visible lines in widget
-            # Else, loop range will be last_line + 1 (default)
-                _range = last_line + max_lines + 1 if self.tilde is not None else last_line + 1
-        
+            # Saving all wrapped lines occurrences
+            wrapped_lines = 0
+
+            # Check if textwidget has the count method (CTkTextbox does not have it)
+            has_count_method = hasattr(self.textwidget, "count") and callable(
+                getattr(self.textwidget, "count")
+            )
+
         # Draw the line numbers looping through the lines
-        for lineno in range(first_line, _range):
+        for lineno in range(first_line, first_loop_range):
+            # If user is using tilde chars, it is necessary to know how many wrapped lines
+            if self.tilde:
+                # If user is using tkinter Text class
+                if has_count_method:
+                    is_wrapped = self.textwidget.count(
+                        f"{lineno}.0", f"{lineno}.0 lineend", "displaylines"
+                    )
+                # If user is using customtkinter and textwidget is a CTkTextbox instance
+                else:
+                    is_wrapped = self.textwidget._textbox.count(
+                        f"{lineno}.0", f"{lineno}.0 lineend", "displaylines"
+                    )
+
+                if is_wrapped:
+                    wrapped_lines += is_wrapped[0]
+
             # Check if line is elided
```

Comment:
Lets try and add this or explain why it was removed ;)

Reply:
I added this wrapped_lines variable to count how many wrapped lines exist before adding tilde chars, so they can be in the correct line counter position